internal/assets/token_pool.go

```diff
@@ -123,7 +132,7 @@ func (am *assetManager) createTokenPoolInternal(ctx context.Context, pool *core.
 		return nil, err
 	}
 
-	_, err = am.operations.RunOperation(ctx, opCreatePool(newOperation, &pool.TokenPool))
+	_, err = am.operations.RunOperation(ctx, opCreatePool(newOperation, &pool.TokenPool), pool.IdempotencyKey != "")
```

Comment:
I'm trying to page back in to why the IdempotencyKey here lives on the pool (and the transfer, and the approval) instead of on the operation or the transaction...

Reply:
Never mind; this is just the input DTO. It is eventually stored on the transaction.